src/actions-util.ts

```diff
@@ -425,6 +425,36 @@ export async function getAnalysisKey(): Promise<string> {
   return analysisKey;
 }
 
+export async function getAutomationID(): Promise<string> {
+  const analysis_key = await getAnalysisKey();
+  const environment = getRequiredInput("matrix");
+
+  return computeAutomationID(analysis_key, environment);
+}
+
+export function computeAutomationID(
+  analysis_key: string | undefined,
+  environment: string | undefined
+): string {
+  let automationID = `${analysis_key}/`;
+
+  // the id has to be deterministic so we sort the fields
+  if (environment !== undefined && environment !== "null") {
```

Comment:
The function parameter type does not permit `null` for the `environment`. Either the type specification is incorrect, or you don't need to check `environment !== "null"`. But maybe a simpler solution would be:

```suggestion
  if (environment) {
```

This would also avoid an error on an empty string.

Reply:
I did change this, but rolled back because it was indeed needed https://github.com/github/codeql-action/pull/463/checks?check_run_id=2508425922. I don't know why, but sometimes we get `environment = "null"`

Reply:
Hmmm...that's very odd. OK. I guess we need to keep it.